src/test/java/com/google/sps/CommentAnalysisTest.java

```diff
@@ -97,25 +112,76 @@ public void testSentimentAnalysisInRange() {
 
   @Test
   public void testCategorizationEdgeCases() {
-    // Test with mocked analysis interface
-    Assert.assertEquals(NORMAL_STAT.getAggregateValues().get(new Range(0, 0.2)).intValue(), 4);
-    Assert.assertEquals(EDGE_STAT.getAggregateValues().get(new Range(-1.0, -0.8)).intValue(), 1);
-    Assert.assertEquals(EDGE_STAT.getAggregateValues().get(new Range(0.8, 1)).intValue(), 1);
-    // Test without mocked analysis interface
-    Assert.assertEquals(NORMAL_STAT.getAggregateValues().get(new Range(-0.2, 0)).intValue(), 2);
     Assert.assertEquals(
-        ALL_OUSIDE_STAT.getAggregateValues().get(new Range(-1.0, -0.8)).intValue(), 0);
+        EDGE_STAT
+            .getAggregateValues()
+            .get(new Range(BigDecimal.valueOf(-1.0), BigDecimal.valueOf(-0.8)))
+            .intValue(),
+        1);
+    Assert.assertEquals(
+        EDGE_STAT
+            .getAggregateValues()
+            .get(new Range(BigDecimal.valueOf(0.8), BigDecimal.valueOf(1.0)))
+            .intValue(),
+        1);
+  }
+
+  @Test
+  public void testCategorizationNormalCases() {
+    Assert.assertEquals(
+        NORMAL_STAT
+            .getAggregateValues()
+            .get(new Range(BigDecimal.valueOf(0.0), BigDecimal.valueOf(0.2)))
+            .intValue(),
+        4);
     Assert.assertEquals(
-        ALL_OUSIDE_STAT.getAggregateValues().get(new Range(0.8, 1.0)).intValue(), 0);
+        NORMAL_STAT
+            .getAggregateValues()
+            .get(new Range(BigDecimal.valueOf(-0.2), BigDecimal.valueOf(0.0)))
+            .intValue(),
+        2);
+  }
+
+  @Test
+  public void testCategorizationOutsiderCases() {
     Assert.assertEquals(
-        ONE_OUSIDE_STAT.getAggregateValues().get(new Range(-1.0, -0.8)).intValue(), 1);
+        ONE_OUSIDE_STAT
+            .getAggregateValues()
+            .get(new Range(BigDecimal.valueOf(-1.0), BigDecimal.valueOf(-0.8)))
+            .intValue(),
+        1);
   }
 
   @Test
-  public void testAvgScore() {
-    Assert.assertEquals(EDGE_STAT.getAverageScore(), 0.0, 0);
+  public void testAvgNormalScore() {
     Assert.assertEquals(SYMMETRIC_STAT.getAverageScore(), 0.0, 0);
-    Assert.assertEquals(ALL_OUSIDE_STAT.getAverageScore(), -99, 0);
+  }
+
+  @Test
+  public void testAvgEdgeScore() {
+    Assert.assertEquals(EDGE_STAT.getAverageScore(), 0.0, 0);
+  }
```

Comment:
I would combine these with testCategorizationEdgeCases as your input is the same. With the global variables removed, it makes even more sense to combine them :)